### PR TITLE
store: stop updating pessimistic transaction's lock TTL when the session is killed

### PR DESCRIPTION
**Pull request by tiancaiamao**
### What problem does this PR solve? 

```
mysql> show create table t1;
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| Table | Create Table                                                                                                                                                                      |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
| t1    | CREATE TABLE `t1` (
  `id` int(11) NOT NULL,
  `b` int(11) DEFAULT NULL,
  PRIMARY KEY (`id`),
  UNIQUE KEY `b` (`b`)
) ENGINE=InnoDB DEFAULT CHARSET=utf8mb4 COLLATE=utf8mb4_bin |
+-------+-----------------------------------------------------------------------------------------------------------------------------------------------------------------------------------+
1 row in set (0.00 sec)

# session1:
mysql> begin pessimistic;                                                                                                                                                                             Query OK, 0 rows affected (0.01 sec)

mysql> update t1 set b = 3 where id = 1;                                                                                                                                                              
Query OK, 1 row affected (0.01 sec)
Rows matched: 1  Changed: 1  Warnings: 0

# session2:
mysql> update t1 set b = b + 1 where id = 1;   // block here

# session3:
mysql> show processlist;
+------+------+-----------+------+---------+------+-------+--------------------------------------+
| Id   | User | Host      | db   | Command | Time | State | Info                                 |
+------+------+-----------+------+---------+------+-------+--------------------------------------+
|    3 | root | 127.0.0.1 | test | Query   |    0 | 2     | show processlist                     |
|    1 | root | 127.0.0.1 | test | Sleep   |   62 | 3     | NULL                                 |
|    2 | root | 127.0.0.1 | test | Query   |   10 | 2     | update t1 set b = b + 1 where id = 1 |
+------+------+-----------+------+---------+------+-------+--------------------------------------+
3 rows in set (0.00 sec)

mysql> kill tidb query 1;
Query OK, 0 rows affected (0.00 sec)
```

I expect after kill session 1, the session2 would continue, but get:

```
ERROR 9004 (HY000): Resolve lock timeout
```

### What is changed and how it works?

Current kill implementation is setting a `killed` flag and let the killed session to handle it.
Here the session1 is idle, so it doesn't check the `killed` flag. Then the pessimistic lock is still there,
and session2 is blocked by that lock.

I make that `ttlManager` exit when it sees the `killed` flag, so the pessimistic lock could be resolved.

In this PR, I also change the pessimistic lock TTL to a small value and don't use the `PessimisticLockTTL`, because ttlManager can update the TTL.

There is another small change in mocktikv when handling Prewrite.
If the lock TTL is greater than the request's TTL, do not decrease the TTL value.

### Check List 

Tests 

 - Unit test
 - Manual test (add detailed scripts or steps below)

Related changes

 - Need to cherry-pick to the release branch

Release note

 - Write release note for bug-fix or new feature.

### CODE REVIEW

**Comment by tiancaiamao** [comment]
PTAL @coocood

File: store/tikv/2pc.go [hunk 1]
```diff
@@ -598,14 +599,18 @@ func (tm *ttlManager) close() {
 }
 
 func (tm *ttlManager) keepAlive(c *twoPhaseCommitter) {
-	// Ticker is set to 1/3 of the PessimisticLockTTL.
-	ticker := time.NewTicker(time.Duration(PessimisticLockTTL) * time.Millisecond / 3)
+	const lockTTL = 3000 // 3s
```
Comment (coocood) [inline store/tikv/2pc.go]: This is too short.

Comment (tiancaiamao) [reply store/tikv/2pc.go]: Now the value is updated to PessimisticLockTTL

File: config/config.toml.example [hunk 2]
```diff
@@ -320,7 +320,7 @@ max-retry-count = 256
 
 # default TTL in milliseconds for pessimistic lock.
 # The value must between "15s" and "120s".
```
Comment (coocood) [inline config/config.toml.example]: Update the comments

Comment (tiancaiamao) [reply config/config.toml.example]: Done

**Comment by coocood** [comment]
LGTM

**Comment by tiancaiamao** [comment]
PTAL @jackysp

**Comment by tiancaiamao** [comment]
PTAL @jackysp @lysu

**Comment by coocood** [comment]
@MyonKeminta PTAL

**Comment by crazycs520** [comment]
@tiancaiamao  need cherry-pick to 3.1?

**Pull request merged**